### PR TITLE
add signing unit test

### PR DESCRIPTION
**Pull request by jsun28**
add signing unit test

### CODE REVIEW

**Comment by lukewrites** [comment]
> 
> 
> Hello @jsun28! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:
> 
>     * In the file [`packaging/package/jar_jdk_signer.py`](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/package/jar_jdk_signer.py):
> 
> 
> > [Line 83:1](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/package/jar_jdk_signer.py#L83): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> 
>     * In the file [`packaging/tests/test_jar_signer.py`](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_jar_signer.py):
> 
> 
> > [Line 3:32](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_jar_signer.py#L3): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> > [Line 10:20](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_jar_signer.py#L10): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> > [Line 15:1](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_jar_signer.py#L15): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> > [Line 17:5](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_jar_signer.py#L17): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> > [Line 40:1](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_jar_signer.py#L40): [W391](https://duckduckgo.com/?q=pep8%20W391) blank line at end of file
> 
>     * In the file [`packaging/tests/test_package.py`](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_package.py):
> 
> 
> > [Line 26:9](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_package.py#L26): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)

> > [Line 26:134](https://github.com/tableau/connector-plugin-sdk/blob/b4f0e9aa2dfee4c57e5bbfb734c07928a5ba33a1/packaging/tests/test_package.py#L26): [E202](https://duckduckgo.com/?q=pep8%20E202) whitespace before ')'

These are worth fixing.

File: packaging/tests/test_package.py [hunk 1]
```diff
@@ -23,7 +23,8 @@ def test_package_main(self):
             path_to_test_file.unlink()
 
 
-        os.system("python -m package.package " + str(files_directory) + " --dest " + str(expected_dest_directory))
+        os.system("py -3 -m package.package " + str(files_directory) + " --package-only " + " --dest " + str(expected_dest_directory) )
```
Comment (lukewrites) [inline packaging/tests/test_package.py]: I think changing `python` to `py -3` will make this only work on Windows. The dev workflow for `packaging` specifies the use of a venv, IIRC. If the developer is using a venv it should be safe to just call `python` and get python3.7.x

File: packaging/tests/test_jar_signer.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+TEST_FOLDER = Path("tests/test_resources/test_taco_to_sign")
+TEST_TACO = "postgres_odbc.taco"
+TEST_ALIAS_DIFF_PWD = "key_diff_pwd"
+TEST_ALIAS_SAME_PWD= "key_same_pwd"
+TEST_KEYSTORE = "tests/test_resources/test_keystore/test_ks.jks"
+
+
+class TestJarSigner(unittest.TestCase):
```
Comment (lukewrites) [inline packaging/tests/test_jar_signer.py]: I _think_ you can apply the `patch` here and then not have to repeat it for each test method. [Here's the python docs about it](https://docs.python.org/3.5/library/unittest.mock-examples.html#applying-the-same-patch-to-every-test-method)

```suggestion
@patch('package.jar_jdk_signer.get_user_pwd')
class TestJarSigner(unittest.TestCase):
```

I haven't tried the above, so be sure to test.

File: packaging/package/jar_jdk_signer.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 KEYSTORE_PWD_PROMPT_LENGTH = len("Enter Passphrase for keystore: ")
 ALIAS_PWD_PROMPT_LENGTH = len("Enter key password for : ")
 
+
```
Comment (pvanderknyff) [inline packaging/package/jar_jdk_signer.py]: Extra space

**Pull request merged**